dvc/stage/__init__.py

```diff
@@ -296,8 +296,11 @@ def env(self, checkpoint_func=None) -> Env:
 
         for out in self.outs:
             current = self._read_env(out, checkpoint_func=checkpoint_func)
-            if set(env.keys()).intersection(set(current.keys())):
-                raise DvcException("Duplicated env variable")
+            if any(
+                env.get(key) != current.get(key)
+                for key in set(env.keys()).intersection(current.keys())
+            ):
```

Comment:
If we have multiple outs of some type that sets an env var (like `checkpoint`, it is expected that `_read_env` will return the same keys for each out. We should be checking that the values of any repeated keys don't conflict with each other, not that the duplicate keys exist.